core/src/main/scala/za/co/absa/fadb/status/package.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2022 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.fadb
+
+import za.co.absa.fadb.exceptions.StatusException
+
+package object status {
+
+  /**
+    *  Class represents the status of calling a fa-db function (if it supports status that is)
+    */
+  case class FunctionStatus(statusCode: Int, statusText: String)
+
+  /**
+    *  Represents a function status with data.
+    *  @param functionStatus the function status
+    *  @param data the data of one row (barring the status fields)
+    *  @tparam D the type of the data
+    */
+  case class Row[D](functionStatus: FunctionStatus, data: D)
+
+  /**
+    *  This is a representation of a single row returned from a DB function with processed status information.
+    *
+    *  Note: D here represents a single row reduced by status-related columns, i.e. a type of data.
+    */
+  type FailedOrRow[D] = Either[StatusException, Row[D]]
```

Comment:
How about `StatusExceptionOr[A]` - it's more common to have sth like 'ErrorOr[A]' or sth like my suggestion which has the type encoding the successful value in the name.

Reply:
Hmm but sometimes it can happen that the type parameter is provided somehow automatically (is the compiler can understand it from the caller, then I won't need to specify it explicitly), which would lead to odd name

On the other hand, we can always explicitly supply it..

I must admit that I haven't really seen such approach in the past, but then you are much more experienced in Scala - I'll think about it, because the main idea was to make those 3 classes relatable = so I value 3 classes named well enough and relate-able, rather than each class perfectly named but not relate-able. WDYT?